runtime/descriptor.go

```diff
@@ -19,6 +19,13 @@ import (
 	"reflect"
 )
 
+type fieldDescriptorType int
+
+const (
+    FIELDDESCRIPTOR_RO fieldDescriptorType = iota
```

Comment:
Really? This is so ugly and impossible to distinguish constants from variables. Ok... done on 845490d91